packages/azure-http-specs/specs/azure/client-generator-core/client-default-value/main.tsp

```diff
@@ -114,3 +114,30 @@ op getPathParameter(
 
   @path segment2: string,
 ): void;
+
+@scenario
+@route("/header-parameter")
+@scenarioDoc("""
+  Test case 4: `@clientDefaultValue` for header parameters.
+  This scenario tests that client default values are correctly applied to header parameters.
+  
+  Expected header parameters:
+  Accept: "application/json;odata.metadata=none" (default)
+  x-custom-header: "default-value" (default)
+  
+  Expected response: 204 No Content
+  """)
+@get
+op getHeaderParameter(
+  #suppress "@azure-tools/typespec-azure-core/no-legacy-usage" "Testing legacy decorator"
+  @global.Azure.ClientGenerator.Core.Legacy.clientDefaultValue(
+    "application/json;odata.metadata=none"
+  )
+  @header("Accept")
+  accept?: string,
```

Comment:
Unsure whether this would affect TCGC logic or emitter logic on multiple content-type.

Reply:
What I mean is that a settable `accept` could make the op be multiple content-type (meaning, user can set different `accept` to get different content). The value itself does not matter much.

I am fine with it if no other language has concern.

Reply:
Understood. This case may not that proper here for confusion about multi content-types. But Python really need this case to cover our default value logic for special header (e.g. accept). And will wait for other reviewers.